parlai/core/teachers.py

```diff
@@ -324,23 +324,16 @@ def next_batch(self):
 
         return self.batches[batch_idx]
 
-    def num_episodes(self):
+    def num_episodes(self) -> int:
         """
         Get the number of episodes in this dataset.
         """
-        if self.use_batch_act:
-            # when using batch_act, this is length of sorted data
-            return len(self.sorted_data)
         raise RuntimeError('"num_episodes" must be overriden by children.')
 
-    @lru_cache(maxsize=1)
-    def num_examples(self):
+    def num_examples(self) -> int:
         """
         Get the total number of examples in this dataset.
         """
-        if self.use_batch_act:
-            # when using batch_act, this is length of sorted data
-            return len(self.sorted_data)
```

Comment:
note that `sorted_data` was never called anywhere in the codebase. this was an example of a static failure being caught by the type checker. Really not sure where it came from, but better to just require the user implement the method.

Reply:
it's an artifact of the old batchsort, thanks for removing